### DISCUSSION TITLE
[Laptop] Clicks unaligned with mouse cursor how to debug?

### DISCUSSION THREAD

**Discussion by LordXerus** [qtile/qtile#4655] [Q&A]
I was trying some setups with `xrandr --output Laptop --output HDMI --right-of Laptop`

At first it was working but then it caused weird bugs (clicks all seemed to be triggering at the top of the screen, despite the cursor being lower). I think I fixed it by restarting via mod ctrl r, but what is the recommended solution?

**Comment by tych0** [comment]
Qtile doesn't really support changing this on the fly, so you'll have to restart it in order to get it to reason correctly about different monitor orientations.

**Reply by elParaguayo** [reply]
Is that right? We reconfigure screens (including bars and widgets) when there's a change like this. I would have thought this would be enough but I'm often wrong!

**Reply by jrwrigh** [reply]
In my experience, the reconfigure screens is hit-and-miss. Sometimes I'll connect to a screen and qtile will reconfigure without needing to restart, sometimes I do need to restart. I think it's primarily when my `autorandr` kicks in and changes things, but sometimes I'll need to restart even if I manually change the screen layout.

Edit: I should mention the need to restart isn't related to OP's cursor problems. For me it's mainly that the qtile reconfigured screens don't line up with the actual monitors.

**Reply by tych0** [reply]
> Is that right? We reconfigure screens (including bars and widgets) when there's a change like this.

Maybe it's not :). I had forgotten you guys had done a bunch of work in this area, but by experience matches @jrwrigh's, in that it is pretty hit or miss. I always restart myself.

**Reply by elParaguayo** [reply]
Well, it should work as that's the point of reconfiguring... Guess there are still some bugs in that code.

**Comment by Etn40ff** [comment]
> > Is that right? We reconfigure screens (including bars and widgets) when there's a change like this.
> 
> Maybe it's not :). I had forgotten you guys had done a bunch of work in this area, but by experience matches @jrwrigh's, in that it is pretty hit or miss. I always restart myself.

Just to say that I often experience the same thing with 2 monitors and autorand.
